modules/scripts/genmainindex.py

```diff
@@ -0,0 +1,19 @@
+import os
+
+toctree = []
+for (dirpath, dirnames, filenames) in os.walk(os.curdir):
+    caption_template = '   :caption: {caption}'
+    content_template = '   {document}'
+    if 'index.rst' in filenames:
+        caption = dirpath[2:]
```

Comment:
`caption` is the title for the specific toctree. Use the logic from genindex.py. It already handles it. Use `os.path.basename`.